src/tests/plugins/sendmail/test_rules.py

```diff
@@ -0,0 +1,250 @@
+
+    expected = dt_now.replace(hour=4, minute=30, second=0, microsecond=0)
+
+    m = ScheduledMail.objects.filter(rule=rule).first()
+    m.recompute()
+    assert m.computed_datetime.astimezone(utc) == expected
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('send_to,amount_mails,recipients', [
+    (Rule.CUSTOMERS, 1, ['dummy@dummy.test']),
+    (Rule.ATTENDEES, 1, ['meow@dummy.test']),
+    (Rule.BOTH, 2, ['dummy@dummy.test', 'meow@dummy.test']),
+])
+@scopes_disabled()
+def test_sendmail_rule_send_order_vs_pos(send_to, amount_mails, recipients, order, event, item):
+    djmail.outbox = []
+
+    order.status = order.STATUS_PAID
+    order.save()
+
+    order.event.rule_set.create(date_is_absolute=False, offset_is_after=False, send_offset_days=2,
+                                send_offset_time=datetime.time(4, 30), send_to=send_to,
+                                subject='meow', template='meow meow meow')
+    order.all_positions.create(item=item, price=0, attendee_email='meow@dummy.test')
+
+    periodic_task.send(sender=None)  # to trigger sendmail_run_rules
+
+    assert len(djmail.outbox) == amount_mails
+
+    _recipients = [mail.to[0] for mail in djmail.outbox]
+    assert set(recipients) == set(_recipients)
+
+
+@pytest.mark.django_db
+@scopes_disabled()
+def test_sendmail_rule_send_correct_subevent(order, event_series, subevent1, subevent2, item):
+    djmail.outbox = []
+
+    order.status = order.STATUS_PAID
+    order.save()
+
+    event_series.rule_set.create(date_is_absolute=False, offset_is_after=False, send_offset_days=2,
+                                 send_offset_time=datetime.time(9, 30), send_to=Rule.ATTENDEES,
+                                 subject='meow', template='meow meow meow')
+    p1 = order.all_positions.create(item=item, price=13, attendee_email='se1@dummy.test', subevent=subevent1)
+    order.all_positions.create(item=item, price=23, attendee_email='se2@dummy.test', subevent=subevent2)
+
+    periodic_task.send(sender=None)
+
+    assert len(djmail.outbox) == 1
+
+    assert djmail.outbox[0].to[0] == p1.attendee_email
+
+
+@pytest.mark.django_db
+@scopes_disabled()
+def test_sendmail_rule_send_correct_products(event, order, item, item2):
+    djmail.outbox = []
+
+    order.status = order.STATUS_PAID
+    order.save()
+
+    rule = event.rule_set.create(send_date=dt_now - datetime.timedelta(hours=1), send_to=Rule.ATTENDEES,
+                                 subject='meow', template='meow meow meow', all_products=False)
+
+    rule.limit_products.set([item])
+    rule.save()
+
+    p1 = order.all_positions.create(item=item, price=13, attendee_email='item1@dummy.test')
+    order.all_positions.create(item=item2, price=13, attendee_email='item2@dummy.test')
+
+    periodic_task.send(sender=None)
+
+    assert len(djmail.outbox) == 1
+
+    assert djmail.outbox[0].to[0] == p1.attendee_email
+
+
+@pytest.mark.django_db
+@scopes_disabled()
+def test_sendmail_rule_send_correct_order_status(event, order):
```

Comment:
this one tests that include_pending=True works, can you also test that

* orders with status=CANCELED, status=EXPIRED or require_approval=true never get mail
* include_pending=False works?